clients/src/main/java/org/apache/kafka/clients/consumer/KafkaShareConsumer.java

```diff
@@ -115,23 +115,26 @@
  * as naturally happens when the locks time out. This limit is controlled by the broker configuration property
  * {@code group.share.record.lock.partition.limit}. By limiting the duration of the acquisition lock and automatically
  * releasing the locks, the broker ensures delivery progresses even in the presence of consumer failures.
+ *
  * <p>
- * The consumer can choose to use implicit or explicit acknowledgement of the records it processes.
- * <p>If the application calls {@link #acknowledge(ConsumerRecord, AcknowledgeType)} for any record in the batch,
- * it is using <em>explicit acknowledgement</em>. In this case:
+ * The consumer can choose to use implicit or explicit acknowledgement of the records it processes by configuring the
+ * {@link ConsumerConfig#SHARE_ACKNOWLEDGEMENT_MODE_CONFIG} property. If the property is not set, the default mode is <code>"implicit"</code>.
+ *
+ * <p>If the config is set to "explicit", the consumer is using <em>explicit acknowledgement</em>. In this case:
  * <ul>
+ *     <li>The application is expected to acknowledge all the records it received in the batch before the next call to ({@link #poll(Duration)}</li>
+ *     <li> If the application has some unacknowledged records before the next call to ({@link #poll(Duration)}, then the poll()
+ *     throws an {@link IllegalStateException}. This is because in explicit mode, all the records in the batch should be acknowledged before the next call to poll()</li>
  *     <li>The application calls {@link #commitSync()} or {@link #commitAsync()} which commits the acknowledgements to Kafka.
- *     If any records in the batch were not acknowledged, they remain acquired and will be presented to the application
- *     in response to a future poll.</li>
+ *     If any records in the batch were not acknowledged until the next poll(), an {@link IllegalStateException} is thrown.</li>
  *     <li>The application calls {@link #poll(Duration)} without committing first, which commits the acknowledgements to
  *     Kafka asynchronously. In this case, no exception is thrown by a failure to commit the acknowledgement.
- *     If any records in the batch were not acknowledged, they remain acquired and will be presented to the application
- *     in response to a future poll.</li>
+ *     If any records in the batch were not acknowledged, an {@link IllegalStateException} is thrown.</li>
  *     <li>The application calls {@link #close()} which attempts to commit any pending acknowledgements and
  *     releases any remaining acquired records.</li>
  * </ul>
- * If the application does not call {@link #acknowledge(ConsumerRecord, AcknowledgeType)} for any record in the batch,
- * it is using <em>implicit acknowledgement</em>. In this case:
+ * If the application sets the {@code share.acknowledgement.mode} property to "implicit" or does not configure the mode, then
```

Comment:
Maybe `does not set the {@code share.acknowledgement.mode} property to "explicit", then`....